<launch>
	<!-- Global params -->
	<rosparam param="miro_data_load_dir">~/miro/data/zig-zag</rosparam>
	<rosparam param="miro_data_save_dir">~/miro/data/zig-zag_tests/17</rosparam>
	<rosparam param="image_resize_width">115</rosparam>
	<rosparam param="image_resize_height">44</rosparam>
	<rosparam param="image_subsampling">1</rosparam>
	<rosparam param="wait_for_ready">true</rosparam>
	<rosparam param="calibration_file_left" subst_value="True">$(find miro_teach_repeat)/calibration/left_360.yaml</rosparam>
	<rosparam param="calibration_file_right" subst_value="True">$(find miro_teach_repeat)/calibration/right_360.yaml</rosparam>

	<!-- Hold head in place, reset odometry and disable cliff sensors -->
	<node pkg="miro_teach_repeat" type="robot_setup.py" name="miro_setup" launch-prefix="bash -c 'sleep 2; $0 $@' " output="screen">
		<rosparam param="lift">30</rosparam>
		<rosparam param="yaw">0</rosparam>
		<rosparam param="pitch">8</rosparam>
		<rosparam param="reset_odom">true</rosparam>
		<rosparam param="disable_cliff_sensors">true</rosparam>
	</node>

	<!-- Load image files for matching -->
	<node pkg="miro_teach_repeat" type="image_matcher.py" name="miro_image_matcher" output="screen" />

	<!-- Follow the previous poses using odometry, but use images to correct for odometry drift -->
	<node pkg="miro_teach_repeat" type="localiser.py" name="miro_localiser" output="screen" >
		<rosparam param="stop_at_end">true</rosparam>
		<rosparam param="discrete-correction">false</rosparam>
	</node>

	<!-- Drive to goal pose -->
	<node pkg="miro_teach_repeat" type="drive_to_pose_controller.py" name="miro_drive_to_pose" output="screen"/>

	<!-- Show image matching -->
	<node pkg="image_view" type="image_view" name="image_match_debug_view" >
		<remap from="image" to="/miro/match_image_debug" />
	</node>

	<!-- Rosbagging hurts the performance I think??? -->
	<!-- <node pkg="rosbag" type="record" name="rosbagger" args="/miro/control/cmd_vel /miro/control/command /miro/control/goal /miro/control/kinematic_joints /miro/sensors/cam/left/compressed /miro/sensors/cam/right/compressed /miro/sensors/kinematic_joints /miro/sensors/odom /miro/sensors/odom/integrated /miro/sensors/wheel_speed_cmd"/> -->
</launch>